enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/lucene/LuceneFulltext.java

```diff
@@ -17,73 +17,51 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-package org.neo4j.kernel.api.impl.fulltext;
+package org.neo4j.kernel.api.impl.fulltext.lucene;
 
 import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.store.Directory;
 
+import java.io.Closeable;
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 
-import org.neo4j.internal.kernel.api.InternalIndexState;
 import org.neo4j.kernel.api.impl.index.AbstractLuceneIndex;
 import org.neo4j.kernel.api.impl.index.partition.AbstractIndexPartition;
 import org.neo4j.kernel.api.impl.index.partition.IndexPartitionFactory;
 import org.neo4j.kernel.api.impl.index.partition.PartitionSearcher;
-import org.neo4j.kernel.api.impl.index.partition.WritableIndexPartitionFactory;
 import org.neo4j.kernel.api.impl.index.storage.PartitionedIndexStorage;
 import org.neo4j.kernel.api.impl.schema.writer.PartitionedIndexWriter;
+import org.neo4j.storageengine.api.EntityType;
 
-class LuceneFulltext extends AbstractLuceneIndex
+import static java.util.Collections.singletonMap;
+
+public class LuceneFulltext extends AbstractLuceneIndex implements Closeable
```

Comment:
LuceneFulltext copies LuceneSchemaIndex behavior by copy-pasting, it should extend it instead.

Reply:
While I see where you are coming from, there is also a lot of stuff in LuceneSchemaIndex that we do not want, such as verifying uniqueness and dealing with IndexReader. The things that are identical are isOnline, markAsOnline and markAsFailed. Perhaps we should extract those to the AbstractLuceneIndex instead.

Reply:
Extracted.